packages/search-in-workspace/src/browser/search-in-workspace-widget.tsx

```diff
@@ -368,11 +376,20 @@ export class SearchInWorkspaceWidget extends BaseWidget implements StatefulWidge
                 return;
             } else {
                 this.searchTerm = searchValue;
-                this.resultTreeWidget.search(this.searchTerm, (this.searchInWorkspaceOptions || {}));
+                const options: SearchInWorkspaceOptions = {
+                    ...this.searchInWorkspaceOptions,
+                    ignoredFiles: this.getIgnoredFiles()
+                };
+                this.resultTreeWidget.search(this.searchTerm, (options || {}));
             }
         }
     }
 
+    protected getIgnoredFiles(): string[] {
+        const ignoredFilesValue = { ...this.filesPreferences['files.exclude'], ...this.searchInWorkspacePreferences['search.excludeFiles'] };
+        return Object.keys(ignoredFilesValue).filter(key => !!ignoredFilesValue[key]);
+    }
+
```

Comment:
With the current implementation, the filtering is done in the fe. It was done this way as the props are being passed down by the `siw-widget` hence that's where I filtered the keys(to get rid of false values for e.g ```**/.yml: false```). Would it be better to pass down the entire object in the `ignoredFile` as a prop and then filter it in the backend?